### ISSUE TITLE
Unpin `http` version

### ISSUE THREAD

**Issue by ralf-vogel** [supertokens/supertokens-flutter#54]
`http` was pinned down without an explanation why:
https://github.com/supertokens/supertokens-flutter/blob/e6e6a2815d9488373e773e1af7b8f0a7cc8d62a4/pubspec.yaml#L19

This also pins down `google_sign_in_dartio` to a version < 0.3 and trying to upgrade it leads to this issue:

```shell
> flutter pub get
Resolving dependencies... 
Because no versions of cross_file match >0.3.4+1 <0.4.0 and cross_file 0.3.4+1 depends on web ^0.5.0, cross_file ^0.3.4+1 requires web ^0.5.0.
And because http >=1.1.2 <1.2.1 depends on web >=0.3.0 <0.5.0, cross_file ^0.3.4+1 is incompatible with http >=1.1.2 <1.2.1.
And because google_sign_in_dartio >=0.2.3 depends on http ^1.1.2 and supertokens_flutter 0.3.1 depends on http <1.2.0, one of cross_file ^0.3.4+1 or google_sign_in_dartio >=0.2.3 or
  supertokens_flutter 0.3.1 must be false.
And because no versions of supertokens_flutter match >0.3.1 <0.4.0 and my_project depends on cross_file ^0.3.4+1, google_sign_in_dartio >=0.2.3 is incompatible with
  supertokens_flutter ^0.3.1.
So, because my_project depends on both supertokens_flutter ^0.3.1 and google_sign_in_dartio ^0.3.0, version solving failed.


You can try the following suggestion to make the pubspec resolve:
* Consider downgrading your constraint on google_sign_in_dartio: flutter pub add google_sign_in_dartio:'^0.2.2+1'
```

What is the reason for pinning the `http` package down? Can it be unpinned?

**Comment by rishabhpoddar** [comment]
Makes sense! We will make a release today / tomorrow with this fix. Thanks

**Comment by rishabhpoddar** [comment]
This has been fixed in flutter sdk version >= 0.4.0
